src/swap/SwapReviewScreen.tsx

```diff
@@ -119,34 +122,42 @@ export function SwapReviewScreen(props: Props) {
         setShouldFetch(false)
         setFetchError(true)
         dispatch(showError(ErrorMessages.FETCH_SWAP_QUOTE_FAILED))
-        Logger.debug(TAG, 'Error while fetching transactions', error)
+        Logger.debug(TAG, 'Error while fetching swap quote', error)
       },
     }
   )
 
   const submitSwap = () => {
-    // Check for swapInfo prior to submitting swap
-    if (!swapInfo) return
+    // Check for swapResponse prior to submitting swap
+    if (!swapResponse) {
+      dispatch(showError(ErrorMessages.SWAP_SUBMIT_FAILED))
+      Logger.error(TAG, 'No swap response found')
+      return
+    }
+
     // Analytics for swap submission
     ValoraAnalytics.track(SwapEvents.swap_review_submit, {
       toToken,
       fromToken,
       // TODO: Add fee to total when enabled
-      usdTotal: +divideByWei(swapInfo.unvalidatedSwapTransaction.buyAmount).multipliedBy(
-        swapInfo.unvalidatedSwapTransaction.price
+      usdTotal: +divideByWei(swapResponse.unvalidatedSwapTransaction.buyAmount).multipliedBy(
+        swapResponse.unvalidatedSwapTransaction.price
       ),
-      fee: +divideByWei(swapInfo.unvalidatedSwapTransaction.sellAmount).multipliedBy(
+      fee: +divideByWei(swapResponse.unvalidatedSwapTransaction.sellAmount).multipliedBy(
         swapFeeDecimal
       ),
     })
-    // TODO: dispatch swap submission
+    // Dispatch swap submission
+    if (userInput !== null) {
+      dispatch(swapStart({ ...swapResponse, userInput }))
+    }
   }
 
   return (
     <SafeAreaView style={styles.safeAreaContainer}>
       <CustomHeader title={t('swapReviewScreen.title')} left={<BackButton />} />
       <DisconnectBanner />
-      {shouldFetch && swapInfo === null ? (
+      {shouldFetch ? (
```

Comment:
there is an activity indicator on line 183 already, which is also displayed if `shouldFetch` is true? why do we need both?